sdk/src/wallet/operations/balance.rs

```diff
@@ -27,7 +27,8 @@ where
         let protocol_parameters = self.client().get_protocol_parameters().await?;
         let slot_index = self.client().get_slot_index().await?;
 
-        let wallet_data = self.data().await.clone();
+        let wallet_address = self.address().await;
+        let wallet_ledger = self.ledger().await.clone();
```

Comment:
I know it was already there but ledger is something quite massive for a clone... Wonder if the tradeoff of keeping the lock alive is worth it, probably is since it's a read.

Reply:
I agree. I think cloning isn't work it tbh.

Reply:
Yes was done as part of the changes to reduce the possibility of deadlocks

Reply:
yeah, I kept it

Reply:
I will open an issue then, we should definitely find a way to not clone this

Reply:
https://github.com/iotaledger/iota-sdk/issues/2024